messaging/api/views.py

```diff
@@ -0,0 +1,39 @@
+from django.contrib.auth import get_user_model
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAuthenticated
+
+from .serializers import RapidProMessageSerializer
+from ..chat import ChatManager
+from ..models import Thread
+from iogt_users.authentication import RapidProBasicAuthentication
+
+User = get_user_model()
+
+
+class RapidProWebhook(APIView):
+    authentication_classes = [RapidProBasicAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        serializer = RapidProMessageSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        rapidpro_message_id = serializer.validated_data['id']
+        thread_uuid = serializer.validated_data['to']
+        text = serializer.validated_data['text']
+        quick_replies = serializer.validated_data['quick_replies']
+
+        thread = Thread.objects.get(uuid=thread_uuid)
+        # TODO: Decide how to treat each of these potential errors:
+        # - Invalid thread UUID
+        # - channel UUID mismatch
+        # TODO: Extract attachments from messages.
```

Comment:
@istride @dannyparsons Would want to discuss the limitations related to attachments from RapidPro